pkg/registry/experimental/controller/etcd/etcd_test.go

```diff
@@ -82,8 +83,11 @@ var validScale = extensions.Scale{
 }
 
 func TestGet(t *testing.T) {
-	storage, server, si := newStorage(t)
-	defer server.Terminate(t)
+	storage, server, si, destroyFunc := newStorage(t)
+	defer func() {
+		destroyFunc()
+		server.Terminate(t)
+	}()
```

Comment:
@timothysc - defers are called in the reverse order than they appear - so what I wrote would actually mean
"closeClient and then close server".

Personally, I would prefer having two defers than what you wrote (these will be 2 lines instead of 4 lines of code), but I'm not going to spend time on imposing it (unless I persuaded you that 2 lines are better than 4 lines :-) )